src/dsp_tools/utils/excel2json/utils.py

```diff
@@ -284,3 +284,26 @@ def col_must_or_not_empty_based_on_other_col(
         return pd.Series(combined_array)
     else:
         return None
+
+
+def add_optional_columns(df: pd.DataFrame, optional_col_set: set[str]) -> pd.DataFrame:
+    """
+    This function takes a df and a set of columns which may not be in the df, but whose absence could cause errors in
+    the code following.
+    As a quick fix, they are added to the df.
+
+    Args:
+        df: Original df
+        optional_col_set: set of columns that may not be in the df.
+        If they are not, they will be added.
+
+    Returns:
+        The df with the added columns.
+        If all are already there, the df is returned unchanged.
+    """
+    in_df_cols = set(df.columns)
+    if not optional_col_set.issubset(in_df_cols):
```

Comment:
Cool, I just learned that dicts have this handy built-in method. Thank you 😄

Reply:
This is not a dictionary but a set?